File: files/CH-196.R
library(tidyverse)
library(readxl)

path = "files/CH-196 Column Splitting.xlsx"
input = read_excel(path, range = "B2:B8")
test  = read_excel(path, range = "D2:E8")

result = input %>%
  mutate(`ID.1` = str_remove_all(ID, "[0-9]"),
         `ID.2` = str_remove_all(ID, "[A-Z]") %>% as.numeric()) %>%
  select(-ID)

all.equal(result, test, check.attributes = FALSE)
#> [1] TRUE

File: files/CH-196.py
import pandas as pd
import re

path = "CH-196 Column Splitting.xlsx"
input = pd.read_excel(path, usecols="B", skiprows=1, nrows=7)
test = pd.read_excel(path, usecols="D:E", skiprows=1, nrows=7)

input['ID.1'] = input['ID'].apply(lambda x: re.sub(r'[0-9]', '', x))
input['ID.2'] = input['ID'].apply(lambda x: int(re.sub(r'[A-Z]', '', x)))
result = input.drop(columns=['ID'])

print(result.equals(test)) # True
